source/GameInterface/Services/Clans/Patches/ClanLeaderChangePatch.cs

```diff
@@ -17,38 +17,32 @@ namespace GameInterface.Services.Clans.Patches
     [HarmonyPatch(typeof(ChangeClanLeaderAction), "ApplyInternal")]
     public class ClanLeaderChangePatch
     {
-        private static readonly AllowedInstance<Clan> AllowedInstance = new AllowedInstance<Clan>();
-
         private static readonly Action<Clan, Hero> ApplyInternal =
             typeof(ChangeClanLeaderAction)
             .GetMethod("ApplyInternal", BindingFlags.NonPublic | BindingFlags.Static)
             .BuildDelegate<Action<Clan, Hero>>();
 
         static bool Prefix(Clan clan, Hero newLeader = null)
         {
-            if (AllowedInstance.IsAllowed(clan)) return true;
+            if (AllowedThread.IsThisThreadAllowed()) return true;
 
             if (PolicyProvider.AllowOriginalCalls) return true;
 
             if (ModInformation.IsClient && clan != Clan.PlayerClan) return false;
 
-            CallStackValidator.Validate(clan, AllowedInstance);
-
             MessageBroker.Instance.Publish(clan, new ClanLeaderChanged(clan.StringId, newLeader.StringId));
 
             return false;
         }
         public static void RunOriginalChangeClanLeader(Clan clan, Hero newLeader = null)
         {
-            using (AllowedInstance)
+            GameLoopRunner.RunOnMainThread(() =>
             {
-                AllowedInstance.Instance = clan;
-
-                GameLoopRunner.RunOnMainThread(() =>
+                using(new AllowedThread())
                 {
                     ApplyInternal.Invoke(clan, newLeader);
-                }, true);
-            }
+                }
+            }, true);
```

Comment:
Can you update this to non-blocking in your branch so we dont forget
```suggestion
            });
```